File: backend/internal/domain/models/rotation_staff_branch_position.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// RotationStaffBranchPosition represents a mapping allowing a rotation staff member to work in a branch position
type RotationStaffBranchPosition struct {
	ID                uuid.UUID `json:"id" db:"id"`
	RotationStaffID   uuid.UUID `json:"rotation_staff_id" db:"rotation_staff_id"`
	BranchPositionID  uuid.UUID `json:"branch_position_id" db:"branch_position_id"`
	RotationStaff     *Staff    `json:"rotation_staff,omitempty"`
	BranchPosition    *Position `json:"branch_position,omitempty"`
	SubstitutionLevel int       `json:"substitution_level" db:"substitution_level"` // 1 = preferred, 2 = acceptable, 3 = emergency only
	IsActive          bool      `json:"is_active" db:"is_active"`
	Notes             string    `json:"notes,omitempty" db:"notes"`
	CreatedAt         time.Time `json:"created_at" db:"created_at"`
	UpdatedAt         time.Time `json:"updated_at" db:"updated_at"`
}

File: frontend/src/lib/api/rotation-staff-branch-position.ts
import apiClient from './client';
import { Position } from './position';
import { Staff } from './staff';

export interface RotationStaffBranchPosition {
  id: string;
  rotation_staff_id: string;
  branch_position_id: string;
  rotation_staff?: Staff;
  branch_position?: Position;
  substitution_level: number; // 1 = preferred, 2 = acceptable, 3 = emergency only
  is_active: boolean;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateRotationStaffBranchPositionRequest {
  rotation_staff_id: string;
  branch_position_id: string;
  substitution_level: number;
  is_active?: boolean;
  notes?: string;
}

export interface UpdateRotationStaffBranchPositionRequest {
  substitution_level: number;
  is_active: boolean;
  notes?: string;
}

export const rotationStaffBranchPositionApi = {
  list: async (params?: { staff_id?: string; position_id?: string }) => {
    const response = await apiClient.get('/rotation-staff-branch-positions', { params });
    return (response.data.mappings || []) as RotationStaffBranchPosition[];
  },
  
  getById: async (id: string) => {
    const response = await apiClient.get(`/rotation-staff-branch-positions/${id}`);
    return response.data.mapping as RotationStaffBranchPosition;
  },
  
  create: async (data: CreateRotationStaffBranchPositionRequest) => {
    const response = await apiClient.post('/rotation-staff-branch-positions', data);
    return response.data.mapping as RotationStaffBranchPosition;
  },
  
  update: async (id: string, data: UpdateRotationStaffBranchPositionRequest) => {
    const response = await apiClient.put(`/rotation-staff-branch-positions/${id}`, data);
    return response.data.mapping as RotationStaffBranchPosition;
  },
  
  delete: async (id: string) => {
    const response = await apiClient.delete(`/rotation-staff-branch-positions/${id}`);
    return response.data;
  },
};
